### ISSUE TITLE
Can't create task while offline

### ISSUE THREAD

**Issue by dianaarnos** [HabitRPG/habitica-android#475]
Whenever I have no web connection I can't create tasks.
I enter the screen to create a new task, I write all I need and when I tap on "save" I get a "Connection Error - You are note connected to the internet" and the task is not created.
I don't know if this should be the app's default behaviour, but it would be nice to manage tasks while offline.

**Comment by andrewlord1990** [comment]
Currently, this is done with a web request. Then on success the task is saved to the database. Changing this would require creating locally and saving to DB, then syncing that with the server.

**Comment by cyotee** [comment]
Cool. So do it. Everyone else does. If you have offline sync, I would actually subscribe. Until then, it's useless to me as I can't update it while on a plane without paying for the access.

**Comment by vIiRuS** [comment]
Unfortunately, implementing offline sync in habitica is not as trivial as
with many other todo apps. When you check off a task, you receive
experience, gold and maybe an item. Those calculations would all have to be
implemented, tested and maintained in the app additionally to the server
version. Then there is also the problem with dailies. If a daily is checked
off locally but not synced before your day turns over, what should happen
then? And what if you p.e. Level up using a different client and then use
the mobile app in offline mode to check off tasks. Even if the calculations
for the reward drop are the same as on the server (again, that would have
to be maintained and kept in sync) it would then use old data, because you
still have the lower level.

Since Habitica is open source, feel free to implement a working offline
mode and submit a PR. I’ll happily merge it in.

-- 
Phillip Thelen
Sent with Airmail

**Comment by LunarGem** [comment]
Perhaps actions to be performed server side could be stored in some sort of array, and then exp and gold could be calculated when connection is restored as long as the user is aware the numbers haven't been updated for however long. Is that practical?

**Comment by Mikaela** [comment]
I guess my issue would be a duplicate of this one?

I have a goal of disconncting from networks entirely at 21 if I am at home, so I have time to do my evening tasks such as eating supper and taking a shower. If I don't do that, then I notice that I have been spending a long time online in various communication channels.

My issue is that Habitica only throws me warnings about not being connected when I have disabled mobile data and powered off my WLAN AP, instead of letting me mark good/bad habits or completing tasks.

At one Discord "server" I was told that I could set custom day start at 21 and mark dailies at that time which I did, but I thought that I should report this anyway if it wasn't already done.
